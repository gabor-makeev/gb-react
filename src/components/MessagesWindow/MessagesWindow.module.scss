.messages-window {
  width: 100%;
  padding: 45px 35px;
  &__container {
    display: flex;
    flex-direction: column;
    align-items: center;
    max-width: 1000px;
    margin: 0 auto;
    height: 100%;
    box-shadow: 0 0 5px 2px #131313;
    background-color: #1E1E1E;
    border-radius: 25px;
    padding: 30px 44px;
  }
}

@media (max-width: 768px) {
  .messages-window {
    display: none;
  }

  .active-messaging__messages-window {
    display: flex;
  }
}