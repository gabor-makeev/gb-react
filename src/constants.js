export const AUTHORS = {
  user: 'Gabor',
  bot: 'ðŸ¤– Bot',
};

export const STYLES = {
  color: {
    primary: '#FF4F4F',
    secondary: '#31E865',
  },
};
