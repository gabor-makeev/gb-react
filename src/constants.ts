import { Authors, Messages } from './default-types';

export const api = 'https://api.spaceflightnewsapi.net/v3/articles';

export const NAVIGATION = [
  {
    id: 1,
    name: 'Home',
    path: '/',
  },
  {
    id: 2,
    name: 'Profile',
    path: '/profile',
  },
  {
    id: 3,
    name: 'Messenger',
    path: '/messenger',
  },
];

export const DUMMY_CONTENT = {
  messages: [
    {
      body: 'placerat duis ultricies lacus sed turpis tincidunt id aliquet risus',
      userEmail: Authors.USER,
      userName: Authors.USER,
    },
    {
      body: 'et molestie ac feugiat sed lectus vestibulum mattis ullamcorper velit',
      userEmail: Authors.USER,
      userName: Authors.USER,
    },
    {
      body: 'felis eget velit aliquet sagittis id consectetur purus ut faucibus',
      userEmail: Authors.USER,
      userName: Authors.USER,
    },
  ] as Messages,
};
