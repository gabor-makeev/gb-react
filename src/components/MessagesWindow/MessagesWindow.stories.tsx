import { MessagesWindow } from './MessagesWindow';
import { DUMMY_CONTENT } from '../../constants';
import { ComponentMeta, ComponentStory } from '@storybook/react';

export default {
  title: 'MessageWindow',
  component: MessagesWindow,
} as ComponentMeta<typeof MessagesWindow>;

const Template: ComponentStory<typeof MessagesWindow> = (args) => (
  <MessagesWindow {...args} messages={DUMMY_CONTENT.messages} />
);

export const Primary = Template.bind({});

export const SmallGaps = Template.bind({});
SmallGaps.args = {
  gapType: 'small',
};

export const BigGaps = Template.bind({});
BigGaps.args = {
  gapType: 'big',
};

export const RoundishAndle = Template.bind({});
RoundishAndle.args = {
  angleType: 'roundish',
};

export const SecondaryBackgroundColor = Template.bind({});
SecondaryBackgroundColor.args = {
  backgroundColor: 'secondary',
};
