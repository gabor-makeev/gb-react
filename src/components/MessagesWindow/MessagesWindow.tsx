import style from './MessagesWindow.module.scss';
import { MessageList } from '../StyledMUIComponents/MessageList';
import { Message } from './components/Message/Message';
import { MessageItem } from 'src/default-types';
import { FC } from 'react';
import classNames from 'classnames';

interface MessageWindowProps {
  messages?: MessageItem[];
  gapType?: string;
  angleType?: string;
  backgroundColor?: string;
}

export const MessagesWindow: FC<MessageWindowProps> = ({
  messages = [],
  gapType = 'medium',
  angleType = 'round',
  backgroundColor = 'primary',
}) => {
  const classes = classNames(
    style[`gap-${gapType}`],
    style[`angle-${angleType}`],
    style[`background-color-${backgroundColor}`]
  );

  return (
    <MessageList className={classes} data-testid={'messageList'}>
      {messages.map((message, idx) => (
        <Message message={message} key={idx} />
      ))}
    </MessageList>
  );
};
