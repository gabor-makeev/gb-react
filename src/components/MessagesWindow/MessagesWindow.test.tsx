import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { MessagesWindow } from './MessagesWindow';
import { AUTHORS, DUMMY_CONTENT } from '../../constants';

describe('MessageList', () => {
  it('should render', () => {
    render(<MessagesWindow messages={DUMMY_CONTENT.messages} />);
  });

  it(`should render ${DUMMY_CONTENT.messages.length} messages`, () => {
    render(<MessagesWindow messages={DUMMY_CONTENT.messages} />);

    expect(screen.getAllByText(AUTHORS.user).length).toBe(
      DUMMY_CONTENT.messages.length
    );
  });

  it('should render without messages passed', () => {
    render(<MessagesWindow />);
  });

  it('should render an 1 unordered list', () => {
    render(<MessagesWindow />);

    expect(screen.getAllByTestId('messageList').length).toBe(1);
  });

  // the code displayed below has to be deleted as soon as there is no way to test non-inline styles with jest:

  // it('should render with default style applied', () => {
  //   const messageWindow = render(<MessagesWindow />);
  //
  //   console.log(messageWindow.classes);
  //   expect(screen.getByTestId('messageList')).toHaveStyle('gap: 10px');
  // });

  // the code displayed below has to be deleted as soon as there is no way to test non-inline styles with jest:

  // it('should render with "small" gap type argument passed', () => {
  //   render(<MessagesWindow gapType={'small'} />);
  //
  //   expect(screen.getByTestId('messageList')).toHaveStyle('gap: 5px');
  // });

  it('should render with snapshot', () => {
    const { asFragment } = render(
      <MessagesWindow messages={DUMMY_CONTENT.messages} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
