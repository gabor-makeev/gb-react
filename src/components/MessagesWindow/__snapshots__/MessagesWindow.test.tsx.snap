// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList should render with snapshot 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root MuiList-padding gap-medium angle-round background-color-primary css-li95ps-MuiList-root"
    data-testid="messageList"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding message css-1s94d1u-MuiListItem-root"
    >
      placerat duis ultricies lacus sed turpis tincidunt id aliquet risus
      <span
        class="message__author-sign"
      >
        Gabor
      </span>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding message css-1s94d1u-MuiListItem-root"
    >
      et molestie ac feugiat sed lectus vestibulum mattis ullamcorper velit
      <span
        class="message__author-sign"
      >
        Gabor
      </span>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding message css-1s94d1u-MuiListItem-root"
    >
      felis eget velit aliquet sagittis id consectetur purus ut faucibus
      <span
        class="message__author-sign"
      >
        Gabor
      </span>
    </li>
  </ul>
</DocumentFragment>
`;
